### DISCUSSION TITLE
Proposal: Modlists need allowlists

### DISCUSSION THREAD

**Discussion by temujin9** [bluesky-social/atproto#1978] [Bluesky Lexicons]
**Is your feature request related to a problem? Please describe.**

Modlists as implemented have a gap in usefulness: either they're too small to matter much, or they contain what potential subscribers consider false positives.

**Describe the solution you'd like**

A way of subscribing to a list that is the mirror of a modlist: preventing muting or blocking by modlists for the subscriber. It should not affect personal mutes or blocks. This is separate from following, or subscribing to a list's feed.

**Describe alternatives you've considered**

We might instead (or additionally) allow individuals to break through modlists with a follow. That presumes they want that person's content on their following feed, instead of only when they encounter it on feeds. It also doesn't provide symmetry in shareability, which is necessary for fairness and to reduce required work by targeted groups (as with modlists).

_I'm glad to contribute code to this, if you agree it's worthwhile to work on, and likely to be accepted once written._

**Comment by bnewbold** [comment]
I agree with the motivation here: having to take a modlist all-or-nothing can be difficult. Being able to compose multiple lists (additive/subtractive) could expand the number of folks interested in a list, and make this form of moderation more collaborative and customizable.

The biggest downside I can see is complexity and cognitive load: it becomes harder to figure out whether and why content is showing up in various contexts. It feels like some stacking of prioritization would be necessary. Most of the implementation of modlists is in the backend (appview), and we already have had some concerns about performance and scaling (especially with many large modlists subscribed to by many accounts).

I'm personally hopeful that 3rd party labeling services and other product/protocol interventions could fill the gap here. The behaviors, UX, and capabilities with those will be different, but could work better than large modlists.

What are some ways we could achieve the underlying goal here without a complex mechanism for composing lists natively? Lists could be "forks" (duplicated) and edited. An external tool could make it easier to maintain a list which is derivative of one or more other list (eg, a bot that maintains list C as A-minus-B). Maintainers of large lists could split their lists up in to multiple smaller more granular lists (it is relatively easy for folks to follow multiple lists).

**Reply by pfrazee** [reply]
I do think that having the user's follows override a modlist would make sense?

**Reply by temujin9** [reply]
Unfortunately, it's the part I like the least because it runs over a use-case I have: muting someone I follow temporarily, until they've gotten the topic I dislike out of their system. I could move that to basic mutes, though, so it's not a deal-breaker.

**Reply by temujin9** [reply]
To Bryan's points:

Splitting the lists doesn't really address the core problem: partial mismatch between the list-maker's judgement and list-user's.

External tools might do it, but having to go external would be added friction for both users and maintainers. I suppose I could find personal value in an external tool that lets me say "my list X is user A's list B plus user C's list D, minus the people I've written exceptions for", but I fear the average user couldn't locate it to make use of it. I do see the argument that it's as likely to confuse them and be unused in the app as outside it.

I feel like the stacking order should be a set and immutable thing. I can see someone if I don't have them individually blocked, if they're not on any subscribed blocklists, or if they are on a blocklist but also on an allowlist, and vice versa. Similar (but only outbound) for mutes.

**Reply by temujin9** [reply]
Let me consider what it would take to make the hypothetical "outside the app" tool. I think I can see a path with the Python SDK, and it might at least help highlight why this is better in-app, or (more hopefully) prove you right.

**Reply by omniraptorr** [reply]
this seems like an example of the goal of 'composable moderation' (letting the user layer different moderation policies/lists on top of each other in different ways). Isn't the goal that the end user should be doing a nontrivial amount of the 'composing' in question? it seems the only way to provide both a transparent and customizable experience.

**Comment by temujin9** [comment]
First big problem with the "outside tool" implementation that I can see: it violates the privacy of usage that modlists give, forcing people who want to customize to disclose the lists they subscribe to, in making modified duplicates of them. I have a few lists, and I can attest that the potential blowback for negative list-making is pretty strong. I am comfortable with it, but I'm not typical in that regard.

**Reply by bnewbold** [reply]
This is a great point!

**Reply by temujin9** [reply]
There's a privacy-preserving possibility, but it would be messy. All of the lists could be held in the service's bsky account, instead of the user's. That may have performance implications.

The second problem I see is that the outside tool is very use-specific. Inside-the-app allowlist might also extend to moderation tags, e.g. bumping the moderation level up one for someone on the list (from Hide to Warn, or Warn to Show). This would have the same benefit as with modlists, of making the base idea more useful and palatable.

**Comment by temujin9** [comment]
I've given it some time to percolate, and I don't see a great outside path for this. Are you interested in my contributing code to provide either a) full allowlist as suggested, or b) using Following as an allowlist?

**Reply by ghost** [reply]
did you ever get the green light to contribute to this feature?

**Reply by temujin9** [reply]
No, I never did. Given the similar issues around labels, I think it's still a worthwhile idea to explore.

**Reply by ghost** [reply]
i think it is a worthwhile to deploy, did you ever start working on this?

 would love to see it added. starting to see a lot more comments on bsky and more issues made around this issue..

**Reply by temujin9** [reply]
I have yet to receive indication from Bsky developers that they would be interested, despite several attempts. If you want to try convincing them, I'm still glad to write it if they express interest.

**Comment by temujin9** [comment]
https://bsky.app/profile/msh.bsky.social/post/3l3q63lkfpk2m contains more discussion about why third party tools are a poor patch to this need.

**Reply by temujin9** [reply]
@bnewbold @pfrazee resurfacing this request, to see if perhaps you now have more interest in it than you did last year.

**Reply by temujin9** [reply]
It also looks like labelers could use similar functionality: https://bsky.app/profile/laurenshof.online/post/3l3smagpj2z2q

**Reply by bnewbold** [reply]
I think we will look at this when we holistically re-visit the entire labeling and mod list system, which probably isn't going to happen before the end of 2024.

**Reply by temujin9** [reply]
Understood. That sounds like it's more something you'll want to handle in-house, but let me know if I can offer any help.

**Reply by temujin9** [reply]
@bnewbold Any update on this? Any chance you have more interest in my taking a swing at it, using the subscriber's follow list as the allowlist?

**Comment by Hoid** [comment]
I think one way to do this that would reduce friction for users would be to have it such that you can override modlist effects within your app by simply either 1) going onto the modlist and clicking an "exclude" button beside their handle (or whatever it shows), or 2) going onto their profile and setting their state the way you want it (unblocked/unmuted/etc.) and behind the scenes this would update an "allowlist" that applies only to that user, and overrides the modlist behavior. so if I have 200 people blocked by a modlist and want to unblock Dave I can either find his name in the mod list and click "exclude" or go to his profile manually and click "unblock" and that will update the allowlist in the background. 
the allowlist would almost be more like a master userModOverrideList which would override all mod lists and be stored alongside the user's preferences. the list of who a specific user blocks is stored in the AppView, right, not the PDS? I hear that the user's preferences are stored in the PDS but I assume the AppView calculates who an individual user has blocked/muted/etc. and caches that on the AppView so it doesn't have to reach out to the PDS that often.

**Reply by temujin9** [reply]
It wasn't what I initially considered, but I like this UX possibility.

**Comment by lamrongol** [comment]
I think moderation list has many problems and it should be replaced with Ozone labeler service.
Because you can set labeler to see only "Warning", not mute/block like following image.
![image](https://github.com/user-attachments/assets/8fddfcaf-f46b-4432-8947-1f75eb00a198)

issues which says moderation list has a problem:
- https://github.com/bluesky-social/social-app/issues/5536
- https://github.com/bluesky-social/social-app/issues/3186
- https://github.com/bluesky-social/social-app/issues/5199
- https://github.com/bluesky-social/social-app/issues/2843

**Reply by bszonye** [reply]
i have a similar problem with labelers, though. the "warn" and "hide" options have very similar UX consequences to "mute" and "block" subscriptions with a moderation list. the "show badge" option is less intrusive, but not all labels offer it, and there's still no way to choose between methods on a case-by-case basis.

**Reply by lamrongol** [reply]
OK, but cases which "show badge" option exists is better than moderation list(I edited original post the latest version of "AI Image Labeler", which replaced "warn" to "Show badge")

**Reply by ghost** [reply]
We need allow lists for it to have better function

**Comment by mschwendener** [comment]
> I do think that having the user's follows override a modlist would make sense?

Totally.

**Comment by symmetricalboy** [comment]
Regarding this paragraph in the OP:
> We might instead (or additionally) allow individuals to break through modlists with a follow. That presumes they want that person's content on their following feed, instead of only when they encounter it on feeds. It also doesn't provide symmetry in shareability, which is necessary for fairness and to reduce required work by targeted groups (as with modlists).

There is a separate proposal, #3630, which is exactly this variation, which I believe to be either the better solution, or at least the best first iteration of a solution that could later expand to explicit whitelisting:

> I can see the appeal for the flexibility that a separate allowlist provides, and that's a valid point. There might be some specific edge cases where a user wants to apply a moderation list to an account they also happen to follow.
> 
> However, from a user experience perspective, I strongly agree with the original proposal. For the vast majority of users, the act of following someone is a very clear and intentional declaration: "I want to see this person's content."
> 
> My main concern with a manual, separate allowlist is that it introduces extra steps and complexity that could be confusing, especially for less technical users. As @jeffb4 pointed out, many people are still getting familiar with what modlists even are. Requiring them to manage a separate allowlist—which for most people would likely just mirror their follow list—seems like it could create unnecessary friction.
> 
> <img alt="image" width="276" height="191" src="https://private-user-images.githubusercontent.com/85461393/466089925-90b98c09-1c27-4925-ad36-89d2c726e30d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1MDg0NDQsIm5iZiI6MTc1MjUwODE0NCwicGF0aCI6Ii84NTQ2MTM5My80NjYwODk5MjUtOTBiOThjMDktMWMyNy00OTI1LWFkMzYtODlkMmM3MjZlMzBkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE0VDE1NDkwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNlYjc5MjA4NTcwNzJjZjZjMmM4YmY2MzA4MmQxMzgyYzMxOTYwM2Q0Zjg1N2U4M2Q2OGViZjdmNzVjOTAxMDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.D2ePAsDMDSHBgk3RrawfbQXGdf6BGTHZBlBtbR8ghSw">  
> 
> While a separate list offers more granular control for power users, I wonder if that benefit for a few outweighs the simplicity and intuitive nature of the 'follows override modlists' approach for the broader user base. It seems that for most common scenarios, the original proposal strikes a better balance between powerful functionality and everyday ease of use.

**Comment by temujin9** [comment]
Unfortunately for both proposals, I've (finally) received clear unambiguous communication from the team that [they are not planning to address this soon](https://bsky.app/profile/bnewbold.net/post/3lql5k6ujvs2b), and are also [not interested in outside contributions to provide it](https://bsky.app/profile/bnewbold.net/post/3lrcqisubdk2u). Since this proposal is largely redundant with that one, and I've exhausted my desire to try advancing this one further, I'm going to close this one.

**Reply by symmetricalboy** [reply]
Thanks for the updates/clarifications. Hm.. Perhaps the other proposal will annoy them into doing it? xDD
ty! 🤜💥🤛

**Reply by temujin9** [reply]
Perhaps a proposal less attached to my name will be better received. I can't help but feel this is has been somewhat affected by my ... enthusiastic embrace of mod-lists, and the broadly poor reactions to same.
